tests/acceptance/stepDefinitions/publicLinkContext.js

```diff
@@ -3,50 +3,141 @@ const { When, Then } = require('cucumber')
+})
 
+Then('the fields of the last public link share response of user {string} should include',
+  function (linkCreator, dataTable) {
+    const fieldsData = dataTable.rowsHash()
+    return sharingHelper.assertUserLastPublicShareDetails(linkCreator, fieldsData)
+  })
+
+When('the public uses the webUI to access the last public link created by user {string} with password {string}', async function (linkCreator, password) {
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShareToken = lastShare.token
+  await client.page.publicLinkFilesPage().navigateAndWaitForPasswordPage(lastShareToken)
   return client.page.publicLinkPasswordPage().submitPublicLinkPassword(password)
 })
 
+Then('the user should see an error message on the public link share dialog saying {string}',
+  function (errorMessage) {
+    return client.page.FilesPageElement
+      .publicLinksDialog()
+      .assertErrorMessageForPublicLinks(errorMessage)
+  })
+
+Then('the public link of the user {string} should not have been generated', async function (sharer) {
+  const resp = await sharingHelper.fetchLastPublicLinkShare(sharer)
+  assert.strictEqual(
+    resp, undefined, 'Error with public link shares! ' + resp
+  )
+})
+
+Then('the public should not get access to the publicly shared file', function () {
+  return client
+    .page
+    .publicLinkPasswordPage()
+    .assertResourceAccessDenied()
+})
+
+When('the user tries to edit the public link named {string} of file/folder/resource {string} changing following',
+  async function (linkName, resource, dataTable) {
+    const editData = dataTable.rowsHash()
+    await client.page.FilesPageElement
+      .filesList()
+      .closeSidebar(100)
+      .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
+      .editPublicLink(linkName, editData, false)
+  })
+
+When('the user edits the public link named {string} of file/folder/resource {string} changing following',
+  async function (linkName, resource, dataTable) {
+    const editData = dataTable.rowsHash()
+    await client.page.FilesPageElement
+      .filesList()
+      .closeSidebar(100)
+      .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
+      .editPublicLink(linkName, editData, true)
+  })
+
+When('the user tries to edit expiration of the public link named {string} of file {string} to past date',
+  async function (linkName, resource) {
+    await client.page.FilesPageElement
+      .filesList()
+      .closeSidebar(100)
+      .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
+      .assertDisabledPastExpiryDate(linkName)
+  })
+
+When('the user {string} removes the public link named {string} of file/folder/resource {string} using the webUI',
+  async function (sharer, linkName, resource) {
+    const lastShare = await sharingHelper.fetchLastPublicLinkShare(sharer)
+    lastShareID = lastShare.id
```

Comment:
Where is this used?

Reply:
lastShareID is a global variable! This variable is used in next step of the same scenario!

Reply:
Now it is dismissed. :fire: